src/Containers/MiddleCanvas/Template.tsx

```diff
@@ -0,0 +1,158 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import { 
+    DragDropContext, 
+    Droppable,
+    DroppableProvided,
+    Draggable,
+    DraggableProvided,
+    DraggableStateSnapshot,
+    DropResult,
+} from 'react-beautiful-dnd';
+import { ITemplatePrefill } from './MiddleCanvasTypes';
+import { TableComponent } from './TableComponent';
+import { DroppableElement } from './DroppableElement';
+import { MainInterface, ResponsiveInterface } from '../../Utils/BaseStyles';
+
+export interface TemplateProps extends MainInterface, ResponsiveInterface {
+    isPreview?: boolean,
+    templatePrefills: ITemplatePrefill
+};
+
+const NO_OF_ITEMS_DELETED = 1;
+const REMOVE_NO_ITEMS = 0;
+
+export const Template: React.FC<TemplateProps> = ({
+    isPreview,
+    templatePrefills,
+    ...props
+}): React.ReactElement => {
+    const [items, setItems] = useState<ITemplatePrefill[]>(Object.values(templatePrefills));
+
+    /**
+     * Reorders the draggable elements in a list
+     * @param {ITemplatePrefill[]} list - list of objects to reorder
+     * @param {number} startIndex - index of where the element originates from
+     * @param {number} endIndex - index of where the element will be placed
+     */
+    const reorder = (list: ITemplatePrefill[], startIndex: number, endIndex: number): ITemplatePrefill[] => {
```

Comment:
I need two different onDrag functions, but they are both very similar. I tried to export it and make it general, but it wouldn't really work with typing because list is of different type. Would it be a bad idea to make it an any type? If I make it an or type, it would still give me errors.

Reply:
Have you tried using the typescript
type command? That could solve this problem easily / clean up this code a bit. Your doing inline types for this function which is a bit messy